tests/services/cache-service.test.ts

```diff
@@ -102,13 +102,67 @@ describe("CacheService", () => {
       return Promise.resolve("Schnecke");
     }
 
-    const resultA = cacheService.getForKeyWithLoadingFunction(key, slowLoadingFunction, ttl)
-    resultA.then(_ => {}) // just consume it
+    const result = cacheService.getForKeyWithLoadingFunction(key, slowLoadingFunction, ttl)
+    const handle = result.then(_ => {}) // let result begin to resolve
 
     await sleep(toMilliseconds(cleanupTime))
     cacheService.invalidateOutdated()
 
-    //await sleep(loadingTime * 1000)
+    await handle // wait for result to resolve: no code execution => no exception
     // will fail if getForKeyWithLoadingFunction throws
   }, 1000000) // exaggeratedly high timeout so we can use breakpoints without breaking a sweat
+
+  test("Dead entries do not supplant living ones", async () => {
+    const cacheService = new CacheService()
+    const key = "someKey"
+    // Each must be greater than the previous one. Script may run too slowly for the math to work out for two-digit values.
```

Comment:
Worked most of the time with `<everything> / 10`, but not always - and I only ran it one or two dozen times.

Given that this use case _is_ rather convoluted - myself I was convinced there was _something_ wrong right from the start, but did not, until two hours ago, really grasp _what exactly_ that was, _and_ intermittendly discarded that feeling even though correct - I went for generously high values to eliminate this source for random noise.

... We should probably investigate those [jest timer mocks](https://jestjs.io/docs/timer-mocks) not too far in the future, lest those high-for-a-test values accumulate.
(Then again, I guess there is time for that at least until we set up some sort of CI pipeline, which would actually execute all of them and thence experience such an accumulation.)